wiki/Bazel-Setup-Instructions-for-Mac.md

```diff
@@ -0,0 +1,117 @@
+## Bazel Setup for Mac including Apple silicon (M1/M2) chips
+
+## Instructions
+
+**The steps to install Bazel on Mac are:**
+1. Setup Rosetta Terminal
+2. Install Bazel 
+3. Install the OpenJDK 8
+4. Install Python 2 and make sure it is active in your environment
+5. Set the Android SDK
+6. Set up the environment to be able to build Oppia Android
+7. Verify that the build is working
+8. Run Bazel commands as needed during development
+
+### 1. Setup Rosetta Terminal
+
+- In the Finder app on your Mac, locate the Applications folder from the favorites sidebar.
+- Right-click on your Terminal app and create a duplicate Terminal (and rename it accordingly, say **Terminal Rosetta**, to avoid confusion).
+- On the newly created Terminal Rosetta icon, right-click and select "Get info", and under “General”, check the option "Open using Rosetta".
+
+**Note: Always use the Rosetta terminal for Bazel setup and running the `bash setup.sh` or any Bazel build-related commands.**
+
+### 2. Installing Bazel
+
+1. Install Bazel from [here](https://docs.bazel.build/versions/4.0.0/install-os-x.html#install-with-installer-mac-os-x). Make sure that you follow the instructions for installing a specific version (Oppia Android requires 4.0.0 and won't build on other versions).
+2. After Install Bazel from the above instructions, set the Bazel path permanently in your terminal using these commands:
+    ```
+    sudo nano /etc/paths
+    ```
+   - Enter your password, when prompted.
+   - Go to the bottom of the file, and enter this path (`/Users/{YourMacUserName}/bin`)
+   - Hit control-x to quit.
+   - Enter “Y” to save the modified buffer.
+   - That’s it!  To test it, in a new terminal window, type: `echo $PATH`
+    
+
+3. That’s it, now Bazel is setup permanently in your terminal, you can again check it by using command:
+   ```
+   bazel –version
+   ```
+
+**Note: The path for Bazel 4.0.0 is set before running Bazel build otherwise you get an error.**
+
+### 3. Installing the OpenJDK 8
+
+Oppia Android also requires OpenJDK 8.
+Follow the instructions [here](https://installvirtual.com/install-openjdk-8-on-mac-using-brew-adoptopenjdk/) to install OpenJDK 8. 
+Note that this requires the installation of brew as a pre-requisite, which can be done by following the instructions [here](https://mac.install.guide/homebrew/index.html). You can then set up your `$JAVA_HOME` environment variable using these [instructions](https://stackoverflow.com/a/75167958/11396524).
+
+
+### 4. Installing Python 2
+
+```
+brew install pyenv
+pyenv install 2.7.18
+pyenv global 2.7.18
+```
+
+- To make sure Python 2 is active in your environment, follow these commands in **oppia-android** directory before running the Bazel build.
```

Comment:
...your environment, navigate to the **oppia-android** directory and run the following commands before running the Bazel build.

Reply:
Done